### PR TITLE
 Add updated guides side-nav

### PR DESCRIPTION
**Pull request by belakoszegi**

### CODE REVIEW

File: modules/guides/js/guides_in_page_navigation.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+      parent.appendChild(lastInsertedChild);
+      setParent(i);
+    }
+    return result;
+  },
+  scrollTo: function (id) {
+    if (id) {
+      var scrollToHeading = document.getElementById(id);
+      if (scrollToHeading) {
+        // Subtracting the height of the admin toolbar.
```
Comment (tokapeb) [inline modules/guides/js/guides_in_page_navigation.js:262]: Maybe should be better to check if is this needed at all (non admin user), measure it dynamically (admin is not on top, default size changed manually etc.)
+ Check it on mobile too.

File: modules/guides/css/guides.in-page-navigation.css [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  padding: 0px;
+  text-indent: 0px;
+  list-style-type: none;
+}
+
+#guides__in-page-nav {
+  border-top: 1px solid #ddd;
+}
+
+#guides__in-page-nav .guides__in-page-nav-li {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]: Same here, don't need this specificity.

```suggestion
.guides__in-page-nav-li {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  background-color: #f5f5f2;
+  border: 1px solid #ddd;
+  border-top: 0px;
+  cursor: pointer;
+}
+
+#guides__in-page-nav .guides__in-page-nav-li:hover {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-li:hover {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  background: #fff;
+}
+
+#guides__in-page-nav .guides__in-page-nav-li a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-li a {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  display:inline-block;
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
+  width:100%;
+  color: #434343;
+}
+#guides__in-page-nav .guides__in-page-nav-li .guides__active-nav {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__active-nav {
```

File: modules/guides/src/Controller/GuidesController.php [hunk 6]
```diff
@@ -104,9 +104,10 @@ public function guideContent($filename) {
...
-          'guides/guide'
+          'guides/guide',
+          'guides/in_page_navigation'
         ]
       ]
     ];
   }
-}
\ No newline at end of file
+}
```
Comment (gkurucz) [inline modules/guides/src/Controller/GuidesController.php:114]:
```suggestion
}

```

File: modules/guides/js/guides_in_page_navigation.js [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+      }
+    }
+
+    var lastInsertedChild;
+    var link;
+    var href;
+    for (var i = 0; i < headingList.length; i++) {
+      if (!parent.getAttribute('data-parent-type')) {
+        parent.setAttribute('data-parent-type', headingList[i].nodeName);
+        parent.classList.add('guides__in-page-nav-ul--' + headingList[i].nodeName.toLowerCase());
```
Comment (gkurucz) [inline modules/guides/js/guides_in_page_navigation.js]:
```suggestion
        parent.classList.add('guides__in-page-nav-heading--' + headingList[i].nodeName.toLowerCase());
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  font-weight: bold;
+  text-decoration: underline;
+}
+
+.guides__in-page-nav-ul--h1 a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-heading--h1 a {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  padding-left: 0rem;
+}
+.guides__in-page-nav-ul--h2 a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-heading--h2 a {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  padding-left: 1rem;
+}
+.guides__in-page-nav-ul--h3 a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-heading--h3 a {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  padding-left: 2rem;
+}
+.guides__in-page-nav-ul--h4 a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-heading--h4 a {
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  padding-left: 3rem;
+}
+.guides__in-page-nav-ul--h5 a {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
.guides__in-page-nav-heading--h5 a {
```

File: modules/guides/guides.info.yml [hunk 13]
```diff
@@ -4,3 +4,4 @@ package: Custom
 
 type: module
 core: 8.x
+version: 8.x-1.0
```
Comment (boobaa) [inline modules/guides/guides.info.yml]: Don't add such lines, pls. It's the job of the script that's gonna create the release.

File: modules/guides/guides.libraries.yml [hunk 14]
```diff
@@ -9,3 +9,11 @@ guide:
   css:
     theme:
       css/guides.guide.css: {}
+
+in_page_navigation:
+  version: 1.0
```
Comment (boobaa) [inline modules/guides/guides.libraries.yml:14]: Dunno what this line here means, and I'm not sure if it's needed either. If it's not needed, then pls don't add such lines, as they might be confusing.

**Review by boobaa — CHANGES_REQUESTED** [review]

**Review by tokapeb — APPROVED** [review]

**Review by gkurucz — CHANGES_REQUESTED** [review]
Here are my thoughts.

File: modules/guides/css/guides.in-page-navigation.css [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+    float: right;
+  }
+}
+
+@media screen and (min-width: 768px) {
+  #guides__flex-wrapper {
+    min-width: 20%;
+    margin-right: 1rem;
+  }
+  #guides__in-page-nav.guides__in-page-nav--sticky {
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]: You don't need to be this specific with the selection. We're using BEM to be specific enough with a single class already, so using `.guides__in-page-nav--sticky` is enough.
```suggestion
  .guides__in-page-nav--sticky {
```

File: modules/guides/guides.libraries.yml (hunk 14 above)
Comment (belakoszegi) [reply modules/guides/guides.libraries.yml]: It's the library definition for the in-page-nav.

**Review by boobaa — APPROVED** [review]

File: modules/guides/css/guides.in-page-navigation.css [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+
+.guides__util--hidden {
+  display: none;
+}
+
+.region-content {
+  display: flex;
+}
+
+
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css]:
```suggestion
```

File: modules/guides/css/guides.in-page-navigation.css [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+  background: #fff;
+}
+
+.guides__in-page-nav-li a {
+  display: inline-block;
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
+  width: 100%;
+  color: #434343;
+}
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css:88]:
```suggestion
}

```

File: modules/guides/css/guides.in-page-navigation.css [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+.guides__active-nav {
+  font-weight: bold;
+  text-decoration: underline;
+}
+
+.guides__in-page-nav-heading--h1 a {
+  padding-left: 0rem;
+}
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css:97]:
```suggestion
}

```

File: modules/guides/css/guides.in-page-navigation.css [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+.guides__in-page-nav-heading--h2 a {
+  padding-left: 1rem;
+}
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css:101]:
```suggestion
}

```

File: modules/guides/css/guides.in-page-navigation.css [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+.guides__in-page-nav-heading--h3 a {
+  padding-left: 2rem;
+}
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css:105]:
```suggestion
}

```

File: modules/guides/css/guides.in-page-navigation.css [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+.guides__in-page-nav-heading--h4 a {
+  padding-left: 3rem;
+}
```
Comment (gkurucz) [inline modules/guides/css/guides.in-page-navigation.css:109]:
```suggestion
}

```

**Review by gkurucz — APPROVED** [review]

**Pull request merged**